includes/login-register.php

```diff
@@ -64,63 +70,58 @@ function edd_register_form( $redirect = '' ) {
  * Process Login Form
  *
  * @since 1.0
+ * @since 2.9.24 No longer does validation which would prevent bruteforce detection plugins to be able to integrate.
+ *
  * @param array $data Data sent from the login form
  * @return void
 */
 function edd_process_login_form( $data ) {
+
 	if ( wp_verify_nonce( $data['edd_login_nonce'], 'edd-login-nonce' ) ) {
-		$user_data = get_user_by( 'login', $data['edd_user_login'] );
-		if ( ! $user_data ) {
-			$user_data = get_user_by( 'email', $data['edd_user_login'] );
-		}
-		if ( $user_data ) {
-			$user_ID = $user_data->ID;
-			$user_email = $user_data->user_email;
 
-			if ( wp_check_password( $data['edd_user_pass'], $user_data->user_pass, $user_data->ID ) ) {
+			$user = edd_log_user_in( 0, $data['edd_user_login'], $data['edd_user_pass'], $data['rememberme'] );
 
-				if ( isset( $data['rememberme'] ) ) {
-					$data['rememberme'] = true;
-				} else {
-					$data['rememberme'] = false;
-				}
+			if ( ! $user instanceof WP_User ) {
+					edd_set_error( 'edd_invalid_login', __( 'Invalid username or password.', 'easy-digital-downloads' ) );
+			}
 
-				edd_log_user_in( $user_data->ID, $data['edd_user_login'], $data['edd_user_pass'], $data['rememberme'] );
-			} else {
-				edd_set_error( 'password_incorrect', __( 'The password you entered is incorrect', 'easy-digital-downloads' ) );
+			// Check for errors and redirect if none present
+			$errors = edd_get_errors();
+			if ( ! $errors ) {
+					$redirect = apply_filters( 'edd_login_redirect', $data['edd_redirect'], $user_ID );
+					wp_redirect( $redirect );
```

Comment:
Should this be `wp_safe_redirect` instead?

Reply:
I think it _should_ be but we'll have to do that in a major release. I'll open an issue in 2.10 for it. We can't do that in a minor release as it might break customizations that are using EDD as an SSO or multisite situation.

Reply:
Noted @robincornett https://github.com/easydigitaldownloads/easy-digital-downloads/issues/8087